### PR TITLE
chore: remove references to Modernizr

### PR DESCRIPTION
**Pull request by bsmth**
This PR removes references of Modernizr as a follow-up for the linked PR.

__changes:__
* removing [Detect It](https://github.com/rafgraph/detect-it) in places where we are removing Modernizr, which I assume is a good idea given the reasons for removing Modernizr.
* Add details about `@supports` for CSS feature detection

__TODO:__
- [x] <strike>decision on better feature detection approaches, where to point readers to instead (`@supports`, via API directly).</strike> We have [Implementing feature detection](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Feature_detection) which I am pointing readers to if they want to learn more.

__related issues and pull requests:__
- https://github.com/mdn/content/pull/26853#issuecomment-1595721985

### CODE REVIEW

File: files/en-us/learn/tools_and_testing/cross_browser_testing/feature_detection/index.md [hunk 1]
```diff
@@ -198,145 +199,9 @@ if (window.matchMedia("(max-width: 480px)").matches) {
...
-   }
-   ```
-
-Try your example out! Here we use the `Modernizr.geolocation` test to check whether geolocation is supported by the current browser. If it is, we run some code that gets your device's current location, and plots it on a Google Map.
-
 ## Summary
 
-This article covered feature detection in a reasonable amount of detail, going through the main concepts and showing you how to both implement your own feature detection tests and use the Modernizr library to implement tests more easily.
-
+This article covered feature detection in a reasonable amount of detail, going through the main concepts and showing you how to both implement your own feature detection tests.
```
Comment (bsmth) [inline files/en-us/learn/tools_and_testing/cross_browser_testing/feature_detection/index.md:204]:
```suggestion
This article covered feature detection in a reasonable amount of detail, going through the main concepts and showing you how to implement your own feature detection tests.
```

File: files/en-us/web/progressive_web_apps/responsive/mobile_first/index.md [hunk 2]
```diff
@@ -229,61 +229,8 @@ The second one sets the width of the content at 600px and centers it in the spac
...
-
-Modernizr also puts its feature detect results in a JavaScript `Modernizr` object too, so that you can run JavaScript code selectively depending on feature support. For example, you could do this:
-
-```js
-if (Modernizr.rgba) {
-  // run code that depends on RGBA colors being supported.
-}
-```
+There are dedicated libraries that perform feature detection, but loading polyfills may come at a performance cost to your visitors, especially on older devices, so you should carefully consider if you want to use them.
+An alternative is to write your own feature detection functionality; Mark Pilgrim's [All-In-One Almost-Alphabetical Guide to Detecting Everything](http://diveintohtml5.info/everything.html) is a good place to start.
```
Comment (bsmth) [inline files/en-us/web/progressive_web_apps/responsive/mobile_first/index.md:233]: And I will add https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Feature_detection to this sentence

File: files/en-us/web/api/eventtarget/addeventlistener/index.md [hunk 3]
```diff
@@ -201,8 +201,6 @@ event on the element `someElement`. For the third parameter, if
 `true`; otherwise, we know that we need to pass a Boolean, and we pass
 `false` as the value of the `useCapture` parameter.
 
-If you'd prefer, you can use a third-party library like [Modernizr](https://modernizr.com/docs) or [Detect It](https://github.com/rafgraph/detect-it) to do this test for you.
-
 You can learn more from the article about
 [`EventListenerOptions`](https://github.com/WICG/EventListenerOptions/blob/gh-pages/explainer.md#feature-detection)
 from the [Web Incubator Community Group](https://wicg.github.io/admin/charter.html).
```
Comment (bsmth) [inline files/en-us/web/api/eventtarget/addeventlistener/index.md:206]: TODO: and [JavaScript feature detection](/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Feature_detection)

**Review by sideshowbarker — APPROVED** [review]

**Comment by bsmth** [comment]
Thank you 👍🏻

**Pull request merged**